libs/wingsdk/src/std/resource.ts

```diff
@@ -111,16 +102,73 @@ export interface IResource extends IConstruct, IHostedLiftable {
  */
 export abstract class Resource extends Construct implements IResource {
   /**
-   * Register that the resource type needs to be lifted to the host for the given
-   * operations. A type being lifted to a host means that that type's static members
-   * will be lifted to the host.
+   * A hook called by the Wing compiler once for each inflight host that needs to
+   * use this type inflight. The list of requested inflight methods
+   * needed by the inflight host are given by `ops`.
+   *
+   * This method is commonly used for adding permissions, environment variables, or
+   * other capabilities to the inflight host.
+   */
+  public static onLiftType(host: IInflightHost, ops: string[]): void {
```

Comment:
Could this be marked internal/private? The addition of a new public static method on all resources is a bit distracting IMO.

Reply:
Static inflights feel more like "inflight closures trapped in a namespace", and inflight closures don't have the option of adding associated capabilities either. Given that, it feels a little strange to expose this lifecycle hook to users, but you're right that there wouldn't be another place to do it.

Reply:
Yeah it's true - maybe there's some incidental complexity here from the language's OO model. But I think it might be not too bad, we'll see